### PR TITLE
Control shared/static linking via BUILD_SHARED_LIBS

### PR DESCRIPTION
**Pull request by esteve**
Not ready for review, just submitted this so that it's added to ros2/ros2#306

Connects to ros2/ros2#306

### CODE REVIEW

File: rclcpp/CMakeLists.txt [hunk 1]
```diff
@@ -64,11 +64,12 @@ set(${PROJECT_NAME}_SRCS
   src/rclcpp/utilities.cpp
 )
 
-add_library(${PROJECT_NAME} SHARED
+add_library(${PROJECT_NAME}
   ${${PROJECT_NAME}_SRCS})
 ament_target_dependencies(${PROJECT_NAME}
   "builtin_interfaces"
   "rcl"
+  "rmw"
```
Comment (dirk-thomas) [inline rclcpp/CMakeLists.txt]: Can you please elaborate why `rmw` was added here as well as it being exported.

**Comment by esteve** [comment]
Closing in favor of #444 Thanks @dhood for taking over!

**Comment by dhood** [comment]
didn't mean to shut down the conversation in this thread @esteve; if there are dependencies on `rmw` that you needed in order to build statically it would still be good to understand what was behind it if it's still relevant

**Comment by esteve** [comment]
@dhood sorry I hope I didn't sound harsh, I didn't mean at all. I honestly wanted to thank you for fleshing out this PR, I had totally forgotten about it.

**Comment by dhood** [comment]
nw; I didn't test https://github.com/ros2/rclcpp/pull/444 with static linking, so there's still a chance follow up changes are required, so I'll replace this PR on the higher-level ticket with https://github.com/ros2/rclcpp/pull/444

**Pull request closed without merging**